app/packages/looker/src/worker/indexed-png-decoder.ts

```diff
@@ -0,0 +1,39 @@
+export type ValidPngBitDepth = 1 | 2 | 4 | 8;
+export type ColorPalette = number[][];
+
+export const indexedPngBufferToRgb = (
+  inputData: Uint8Array,
+  bitDepth: ValidPngBitDepth,
+  colorPalette: ColorPalette
+) => {
+  const indicesPerByte = 8 / bitDepth;
+  const numIndices = inputData.length * indicesPerByte;
+  const resultSize = numIndices * 3;
+  const newRgbArray = new Uint8Array(resultSize);
+
+  let rgbOffset = 0;
+
+  // Determine the initial bit mask based on the bit depth:
+  const initialBitMask = ((1 << bitDepth) - 1) << (8 - bitDepth);
+
+  // extract color indexes from data based on bit depth
+  for (const byte of inputData) {
```

Comment:
I think this will be slow. Can this be changed to a regular for loop?

Reply:
It could be more than just the for loop. Just thinking some benchmarking could be done. In my experience a bunch of garbage collection / memory allocation can happen if these for loops over large arrays aren't kept simple.

Reply:
- Good observation, thanks, I changed it to a regular for loop in the next commit.
- So I did some benchmarking...some notes:

1. This input array, because it uses color type 3 encoding, is naturally terse. We're "unpacking" it. The one used in the example has length 115248.
2. I did some poor-man benchmarking in the browser (in the app code) with bit depth = 2 (a middle ground with four colors supported) with 1e5, 1e6, and 1e7 len arrays. Report:

```
Sample size: 115248, average elapsed time: 12.40 ms (REALISTIC INPUT)
Sample size: 1e5, average elapsed time: 11.60 ms (REALISTIC INPUT)
Sample size: 1e6, average elapsed time: 127.20 ms
Sample size: 1e7, average elapsed time: 1779.10 ms
```

Changing to regular for-loop had almost no noticeable impact, perhaps v8's optimization.

Bit depth 1 would be slower but has no practical usage, and bit depths 4 and 8 would be significantly faster.